File: android/core/data/src/main/kotlin/com/aibook/android/core/data/db/OpdsConnectionEntity.kt
package com.aibook.android.core.data.db

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "opds_connections")
data class OpdsConnectionEntity(
    @PrimaryKey
    val id: String,
    val name: String,
    val baseUrl: String,
    val username: String? = null,
    val passwordCiphertext: String? = null,
    val enabled: Boolean = true,
    val lastSyncedAt: Long? = null,
    val bookCount: Int = 0,
    val syncState: String = "IDLE",
    val lastErrorMessage: String? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val syncMode: String = "FULL"
)

File: ios/AiBook/Core/Data/Database/OpdsConnectionEntity.swift
import Foundation
import SwiftData

// MARK: - OpdsConnectionEntity（与安卓 OpdsConnectionEntity 对齐）

@Model
final class OpdsConnectionEntity {
    var id: String = UUID().uuidString
    var name: String = ""
    var baseUrl: String = ""
    var username: String = ""
    var password: String = ""
    var enabled: Bool = true
    var lastSyncedAt: Date?
    var bookCount: Int = 0
    var syncStateRaw: String = OpdsSyncState.idle.rawValue
    var lastErrorMessage: String = ""

    init() {}

    var syncState: OpdsSyncState {
        get { OpdsSyncState(rawValue: syncStateRaw) ?? .idle }
        set { syncStateRaw = newValue.rawValue }
    }

    func toModel() -> OpdsConnection {
        OpdsConnection(
            id: id,
            name: name,
            baseUrl: baseUrl,
            username: username.isEmpty ? nil : username,
            password: password.isEmpty ? nil : password,
            enabled: enabled,
            lastSyncedAt: lastSyncedAt,
            bookCount: bookCount,
            syncState: syncState,
            lastErrorMessage: lastErrorMessage.isEmpty ? nil : lastErrorMessage
        )
    }

    func update(from conn: OpdsConnection) {
        name = conn.name
        baseUrl = conn.baseUrl
        username = conn.username ?? ""
        password = conn.password ?? ""
        enabled = conn.enabled
        lastSyncedAt = conn.lastSyncedAt
        bookCount = conn.bookCount
        syncStateRaw = conn.syncState.rawValue
        lastErrorMessage = conn.lastErrorMessage ?? ""
    }
}
